solr/core/src/java/org/apache/solr/handler/component/ResponseBuilder.java

```diff
@@ -65,12 +65,20 @@
   public boolean doAnalytics;
   public MergeStrategy mergeFieldHandler;
 
+  public String queryID;
+
   private boolean needDocList = false;
   private boolean needDocSet = false;
   private int fieldFlags = 0;
   //private boolean debug = false;
   private boolean debugTimings, debugQuery, debugResults, debugTrack;
 
+  private boolean isCancellation;
+  private String cancellationUUID;
+
+  private boolean isTaskListRequest;
+  private String taskStatusCheckUUID;
```

Comment:
These could be probably limited to just a presence / absence of the UUID, without the additional boolean flag - just check that the corresponding UUID exists, and this will imply that the flag is turned on.